### ISSUE TITLE
A calendar view for events

### ISSUE THREAD

**Issue by Kakirtog** [wwevo/guildhome#79]
It seems that for the website to be clearer, people want a calendar on which you can see when events are happening. Ari suggested using a Google calendar plugin, that can show these, and also offer to directly import them to your personal calendar. My idea is to have about a month into the future visible, and to put it on the Activity/Events pages, on the side underneath the profile stuff.

**Comment by wwevo** [comment]
Since I have never used a calendar and never even looked at the google one, I have to admit that I am not able to integrate anything mentioned in this suggestion. I can set up a calendar view where one month is shown and all events are marked, that's about it :) So a small calendar widget should be no problem, it would then replace the list of upcoming events I suppose.
I just need to know: how is a clumpy calendar block better than a list? it's not like we are going to have ten events running at the same time or anything like that? Ther 'calendar' would be GM every saturday, KAras Gw1 event every sunday and the occasional raid one ^^

**Comment by Kakirtog** [comment]
As mentioned in the other event issue, it's more for the future, for when the website gets more users and events. Also, it's not replacing the block of current events, it's complementing it. The list has more details, so it's clearer what's what, but the calendar makes it easier to see what's when, if events overlap with your schedule, that kind of thing. Dates are useful, but a calendar is clearer, less clutter. And if it's possible to simply magnet them to your own calendar, that's perfect.

**Comment by wwevo** [comment]
Hehe, never had a schedule either :) except for work of course, but that's the same every week *g*
Still, the space below the sidebar is unused anyways, we could try a calendar. 
Ani? Any thoughts on this matter?

**Comment by AniOnGit** [comment]
A calendar, yes, people like calendars and as every community website has one it was kinda expected that this issue will catch our website too ;)
A calendar is nice, but we should make clear what we do want to see in there:
- Every scheduled event or just the ones I signed up for?
- Birthdays?

Thinking about it, I would personally prefer a calendar showing me all scheduled events, so as an event planner I have a better overview about 'time spots' I could choose for a new event. Maybe it would be an idea to leave the list as it is now with a header 'My next events' and then the calendar with 'EoL calendar' showing ALL scheduled events, not just the ones I signed up. That would help me personally the most I guess.

About the transferring to an own calendar, I really don't know how difficult that stuff is to implement, it would again be a nice feature I would use, but it brings more questions to the table like 'how to synchronize' the online event with the one I exported into my calendar. Maybe this part is something to think about in a next step when the basic calendar feature is done?

I can have a look at the mentioned google calendar plugin, just to get an idea.

**Comment by wwevo** [comment]
All I know is, I will not be able to do the integration. I tried to wrap my head around the google auth system and failed miserably. I originally planned to make the website work with google sign-in. I never even understood the documentation :) So if you want this, you'd need to find someone who can do the actual work :)

**Comment by Tucura** [comment]
In my opinion a calendar view is outdated. 
Like this one:
![Outdated calendar](https://cloud.githubusercontent.com/assets/11677107/19684642/b21c4254-9ab8-11e6-8ebc-84d90eb5dae7.png)

I guess a list with the next 5-10 Events would be enough. This list should just show the titles of the events and the time.If somebody is intrested in an event he/she can get details on demand by clicking on it. Maybe looking like this one 

![Fancy list of events](https://cloud.githubusercontent.com/assets/11677107/19684596/5f6dd2fc-9ab8-11e6-8c63-3a700d3a289c.png)

**Comment by AniOnGit** [comment]
I could live with a longer list if we at least have weekday in it for an easier overview (as someone who created events quite regularly ;)).
Would be list then show every scheduled event or just like now the ones I signed up for? If it shows all scheduled events (what I prefer) we could somehow 'highlight' the ones one signed up for.
What about birthdays then, I wouldn't put them in such a list.

**Comment by wwevo** [comment]
Yeah, the weekday would be kinda important. perhaps in the style of
'[Fri] 20:00 Inhouse Night    Dec 12'16'.

What to do with the birthdays though... Perhaps in the members page? Logged in user could get a few extra columns?

**Comment by wwevo** [comment]
just mentioning this: the code (html) needed to display this calendar would by far outweigh the code for the entire website ^^

**Comment by Kakirtog** [comment]
If it's that hard to implement it might not be worth it, but I still think a calendar would be a good idea. I would do 2 calendars; 1 on the activity page, showing events only (or maybe even only the events you signed up for) and if you click on the calendar ypu get taken to a separate calendar page (clicking on the events takes you to the event page). On the ca;endar page birthdays are displayed as well as all events and maybe an option to show regular events like pvp missions. If it's too much trouble to implement maybe we shouldn;t though. I was personally hoping most of the code could be copy-pasted from somewhere.

A list of events would also add to the usability of the website and present events in a clearer fashion, but it's far less clear when compared to a calendar imo.

My idea of a calendar would look something like this (it's a very crude drawing, not at all accurate (notice how missions are suddenly taking place on mondays) but it's just to give an idea):
http://imgur.com/a/nkkyW

There's no big thing of the events, they're limited to just a single word. This word would I think best be the event tupe (missions, pvp, fractals, wvw, raid, fun, prob a couple more). These are presets in the event creation menu. Maybe you can hover over them to see some more data like the first 10 words of the description, the time, who organized it, I'm not sure what exactly. If you click on this 1 word, you get taken to the event page where you can sign up and comment and everything. If you click anywhere else on the calendar, you get taken to a page with the big calendar. Here you can see further ahead than just the next 4 weeks, you can still only see the 1 word of the event (maybe the starting time as well?), but you can also see all the other stuff I mentioned earlier.

All this is of course open to discussion. Please let me know what you think about this!

**Comment by Kakirtog** [comment]
Example of a calendar: http://www.guildoftweakers.org/events
I'm gonna try to see if I can get them to share their code, unless you can extract it in some way.
This website is in Dutch btw, I hope that's not a problem.
